src/stuf/stuf_zds/tests/test_client.py

```diff
@@ -0,0 +1,142 @@
+import os
+
+from django.conf import settings
+from django.core.files import File
+from django.template import Context, Template
+from django.template.loader import render_to_string
+from django.test import TestCase
+
+import requests_mock
+from zgw_consumers.constants import CertificateTypes
+from zgw_consumers.models import Certificate
+
+from ...tests.factories import StufServiceFactory
+from ..client import StufZDSClient
+
+TEST_CERTIFICATES = os.path.join(os.path.dirname(os.path.abspath(__file__)), "data")
+TEST_XML = os.path.join(
+    settings.BASE_DIR,
+    "src",
+    "stuf",
+    "stuf_zds",
+    "templates",
+    "stuf_zds",
+    "soap",
+)
+
+
+@requests_mock.Mocker()
+class StufZdsClientTest(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        with open(
+            os.path.join(TEST_CERTIFICATES, "test.certificate"), "r"
+        ) as certificate_f:
+            with open(os.path.join(TEST_CERTIFICATES, "test.key"), "r") as key_f:
+                cls.client_certificate = Certificate.objects.create(
+                    label="Test client certificate",
+                    type=CertificateTypes.key_pair,
+                    public_certificate=File(certificate_f, name="test.certificate"),
+                    private_key=File(key_f, name="test.key"),
+                )
+                cls.client_certificate_only = Certificate.objects.create(
+                    label="Test client certificate (only cert)",
+                    type=CertificateTypes.cert_only,
+                    public_certificate=File(certificate_f, name="test1.certificate"),
+                )
+                cls.server_certificate = Certificate.objects.create(
+                    label="Test server certificate",
+                    type=CertificateTypes.cert_only,
+                    public_certificate=File(certificate_f, name="test2.certificate"),
+                )
+
+        cls.client_options = {
+            "gemeentecode": "1234",
+            "omschrijving": "my-form",
+            "zds_zaaktype_code": "zt-code",
+            "zds_zaaktype_omschrijving": "zt-omschrijving",
+            "zds_zaaktype_status_code": "zt-st-code",
+            "zds_zaaktype_status_omschrijving": "zt-st-omschrijving",
+            "zds_documenttype_omschrijving_inzending": "dt-omschrijving",
+            "referentienummer": "123-123-123",
+        }
+
+    def test_mutual_tls(self, m):
+        stuf_service = StufServiceFactory.create(
+            soap_service__client_certificate=self.client_certificate,
+            soap_service__server_certificate=self.server_certificate,
+        )
+
+        m.post(
+            stuf_service.soap_service.url,
+            content=render_to_string(os.path.join(TEST_XML, "creeerZaak.xml")).encode(
```

Comment:
`render_to_string` should not take a full/absolute file path, but just the template name. In this case, that would be `stuf_zds/soap/creeerZaak.xml`. We effectively want to decouple from the file system here and just leverage the django template engine.